worker/firewaller/firewaller.go

```diff
@@ -22,16 +23,78 @@ import (
 	"github.com/juju/juju/worker/catacomb"
 )
 
+// FirewallerAPI exposes functionality off the firewaller API facade to a worker.
+type FirewallerAPI interface {
+	WatchModelMachines() (watcher.StringsWatcher, error)
+	WatchOpenedPorts() (watcher.StringsWatcher, error)
+	Machine(tag names.MachineTag) (*firewaller.Machine, error)
+	Unit(tag names.UnitTag) (*firewaller.Unit, error)
+	Relation(tag names.RelationTag) (*firewaller.Relation, error)
+}
+
+// RemoteModelFirewallAPI exposes remote firewaller functionality to a worker.
+type RemoteModelFirewallAPI interface {
+	WatchSubnets() (watcher.StringsWatcher, error)
+}
+
+// RemoteModelFirewallAPICloser implements RemoteModelFirewallAPI
+// and adds a Close() method.
+type RemoteModelFirewallAPICloser interface {
+	io.Closer
+	RemoteModelFirewallAPI
+}
+
+// Environ defines methods to allow the worker in interact with
+// a Juju cloud environment. These are a subset of the methods
+// defined on the environs.Environ interface.
+type Environ interface {
+	environs.Firewaller
+	Instances(ids []instance.Id) ([]instance.Instance, error)
```

Comment:
I think it would be better for this worker to accept two things: an environs.Firewaller, and an Instanceser (ha ha - real name TBD), an each of those just-so-happen to be satisfied by the same environs.Environ.